sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java

```diff
@@ -0,0 +1,190 @@
+  private @Nullable String dsn;
+  private @Nullable String environment;
+  private @Nullable Integer maxBreadcrumbs;
+  private @Nullable Integer shutdownTimeoutMillis;
+  private @Nullable Integer flushTimeoutMillis;
+  private @Nullable Integer readTimeoutMillis;
+  private @Nullable Double sampleRate;
+  private @Nullable Boolean bypassSecurity;
+  private @Nullable Boolean debug;
+  private @Nullable Boolean attachThreads;
+  private @Nullable Boolean attachStacktrace;
+
+  private @NotNull final HubAdapter hubAdapter;
+
+  SentryAppender(@NotNull HubAdapter hubAdapter) {
+    this.hubAdapter = hubAdapter;
+  }
+
+  public SentryAppender() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  public void start() {
+    if (dsn != null) {
+      Sentry.init(
+          options -> {
+            options.setDsn(dsn);
+            Optional.ofNullable(maxBreadcrumbs).ifPresent(options::setMaxBreadcrumbs);
+            Optional.ofNullable(environment).ifPresent(options::setEnvironment);
+            Optional.ofNullable(shutdownTimeoutMillis).ifPresent(options::setShutdownTimeout);
+            Optional.ofNullable(flushTimeoutMillis).ifPresent(options::setFlushTimeoutMillis);
+            Optional.ofNullable(readTimeoutMillis).ifPresent(options::setReadTimeoutMillis);
+            Optional.ofNullable(sampleRate).ifPresent(options::setSampleRate);
+            Optional.ofNullable(bypassSecurity).ifPresent(options::setBypassSecurity);
+            Optional.ofNullable(debug).ifPresent(options::setDebug);
+            Optional.ofNullable(attachThreads).ifPresent(options::setAttachThreads);
+            Optional.ofNullable(attachStacktrace).ifPresent(options::setAttachStacktrace);
+            options.setSentryClientName(BuildConfig.SENTRY_LOGBACK_SDK_NAME);
+            options.setSdkVersion(createSdkVersion(options));
+          });
+    }
+    super.start();
+  }
+
+  @Override
+  protected void append(@NotNull ILoggingEvent eventObject) {
+    hubAdapter.captureEvent(createEvent(eventObject));
+  }
+
+  /**
+   * Creates {@link SentryEvent} from Logback's {@link ILoggingEvent}.
+   *
+   * @param loggingEvent the logback event
+   * @return the sentry event
+   */
+  @SuppressWarnings("JdkObsolete")
+  final @NotNull SentryEvent createEvent(@NotNull ILoggingEvent loggingEvent) {
+    final SentryEvent event = new SentryEvent(new Date(loggingEvent.getTimeStamp()));
+    final Message message = new Message();
+    message.setFormatted(loggingEvent.getFormattedMessage());
+    message.setParams(toParams(loggingEvent.getArgumentArray()));
+    event.setMessage(message);
+    event.setLogger(loggingEvent.getLoggerName());
+    event.setLevel(formatLevel(loggingEvent.getLevel()));
+
+    final ThrowableProxy throwableInformation = (ThrowableProxy) loggingEvent.getThrowableProxy();
+    if (throwableInformation != null) {
+      event.setThrowable(throwableInformation.getThrowable());
+    }
+
+    if (loggingEvent.getThreadName() != null) {
```

Comment:
We actually have a `threads` model which takes a name. Maybe we can add it here though without stacktrace isn't that useful.

We tend to avoid using `extra`. The reason is that it has no "meaning". We'd prefer using `contexts` and setting the key to something more meaningful. 
If not using threads model and thread-name, this being a one-off value, it's OK to leave it on Extra though

Reply:
Ok, got the point. The question is - do we drop it and rely on `attachThreads` or add `Thread` section to `contexts`. To be honest I do see value in having a thread name, but I don't see value for typical backend application to attach list of all threads. Especially when there can be 100 http threads running at once. 

My preference would be to add `Thread` to `contexts` and attach there fields: `id`, `name`.

Reply:
what about we add a single thread to the list then? with id, name? When the attachThreads is false that is